src/Uno.Extensions.Reactive/Collections/Tracking/CollectionAnalyzer.Core.cs

```diff
@@ -0,0 +1,229 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Diagnostics;
+
+namespace Uno.Extensions.Collections.Tracking;
+
+internal partial class CollectionAnalyzer
+{
+	internal static ComparerRef<object?>? GetRef(IEqualityComparer? comparer)
+		=> comparer is null ? null : comparer.Equals;
+
+	internal static ComparerRef<T>? GetRef<T>(IEqualityComparer<T>? comparer)
+		=> comparer is null ? null : comparer.Equals;
+
+	// DEPRECATED: We need to complete extraction of the CollectionUpdater concept
+	internal static CollectionUpdater ToUpdater<T>(Change<T>? changed, ICollectionUpdaterVisitor visitor)
+	{
+		var updaterHead = changed?.ToUpdater(visitor);
+
+		return updaterHead is null
+			? CollectionUpdater.Empty
+			: new(updaterHead);
+	}
+
+	internal static Change<T>? GetChangesCore<T>(
+		RichNotifyCollectionChangedEventArgs arg,
+		Func<IList, ListRef<T>> getRef,
+		ComparerRef<T>? versionComparer)
+	{
+		switch (arg.Action)
+		{
+			case NotifyCollectionChangedAction.Add:
+			case NotifyCollectionChangedAction.Remove:
+			case NotifyCollectionChangedAction.Move:
+			case NotifyCollectionChangedAction.Reset:
+				return new _Event<T>(arg);
+
+			// For the replace, we are digging in the collection to determine if we could simplify the change considering the given comparers
+			case NotifyCollectionChangedAction.Replace:
+				return GetChangesCore(getRef(arg.OldItems), getRef(arg.NewItems), versionComparer, arg.OldStartingIndex);
+
+			default:
+				throw new ArgumentOutOfRangeException(nameof(arg), arg.Action, $"Action '{arg.Action}' not supported.");
+		}
+	}
+
+	internal static Change<T>? GetChangesCore<T>(
+		ListRef<T> oldItems,
+		ListRef<T> newItems,
+		ComparerRef<T>? versionComparer,
+		int eventArgsOffset = 0)
+	{
+		//if (oldItems.Instance is IDifferentialCollection oldDiff
+		//	&& newItems.Instance is IDifferentialCollection newDiff
+		//	&& oldDiff.FindCommonAncestor(newDiff) is { } ancestor
+		//	&& ancestor == oldDiff.Head) // Temp
+		//{
+		//	// 1. Rollback changes from oldDiff to ancestor
+		//	// => Not supported yet, see condition 'ancestor == oldDiff.Head' above
+
+		//	// 2. Apply changes from ancestor to newDiff
+		//	new 
+		//}
```

Comment:
Remove or is a explanation ?

Reply:
Right ... this is for a future PR, will remove it